### PR TITLE
Adding Ability for customer to see/manage items

### PR DESCRIPTION
**Pull request by FMulligan**
Resolves [#8]
Adding Ability for customer to view their items, Add new items, edit/delete when not pickedup/delivered. Added Validity checking for null values (pickup location, delivery location & status). Unit tests have also been added

Tested flow using iOS 17 iPhone 15 simulator & unit tests pass

image

image

image

image

### CODE REVIEW

File: HaulageApplication/HaulageApp/ViewModels/AllItemsViewModel.cs
Comment (MxmIv) [inline HaulageApplication/HaulageApp/ViewModels/AllItemsViewModel.cs:1]: [Nit] This class is quite big, violating the Single Responsibility Principle. 

You should consider extracting navigation logic into a separate service to keep the `AllItemsViewModel` clean and maintainable.

File: HaulageApplication/HaulageApp/ViewModels/AllItemsViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+            AllItems.Add(i);
+        }
+    }
+
+    public int GetLatestTripId()
+    {
+        var tripId = _context.trip
+            .AsQueryable()
+            .OrderByDescending(i => i.Id)
+            .First().Id;
```
Comment (MxmIv) [inline HaulageApplication/HaulageApp/ViewModels/AllItemsViewModel.cs]: I would change this to `FirstOrDefault()` to stop potential errors

Comment (FMulligan) [reply HaulageApplication/HaulageApp/ViewModels/AllItemsViewModel.cs]: Have Changed, this is also further handled in the return

File: HaulageApplication/HaulageAppTests/AllItemsTests.cs
Comment (MxmIv) [inline HaulageApplication/HaulageAppTests/AllItemsTests.cs:1]: you can add tests like  `LoadAllItemsForCustomerAsync`?

Comment (MxmIv) [inline HaulageApplication/HaulageAppTests/AllItemsTests.cs:1]: [nit ]consider adding tests to check for the correct error handling?

File: HaulageApplication/HaulageAppTests/ItemTests.cs [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        _item = new Item
+        {
+            ItemId = 3, PickupLocation = "Loc C", DeliveryLocation = "Loc D", Status = "pending"
+        };
+
+        _mockContext.Setup(e => e.item).Returns(_mockItems.Object);
+    }
+
+    [Fact]
+    public void ShouldAssignItemWhenQureied()
```
Comment (MxmIv) [inline HaulageApplication/HaulageAppTests/ItemTests.cs]: Different naming convention. I would name this like this to make it consistent with others: 
- ApplyQueryAttributes_ShouldAssignItemProperties

File: HaulageApplication/HaulageAppTests/ItemTests.cs
Comment (MxmIv) [inline HaulageApplication/HaulageAppTests/ItemTests.cs:1]: Test coverage expansion: Ensure that methods like Save and Delete handle exceptions correctly. Add tests to simulate scenarios where exceptions are thrown (e.g., database save failures) and verify that the ViewModel behaves as expected (e.g., logging the error).

**Review by MxmIv — CHANGES_REQUESTED** [review]

File: HaulageApplication/HaulageAppTests/AllItemsTests.cs
Comment (FMulligan) [reply HaulageApplication/HaulageAppTests/AllItemsTests.cs:1]: Not adding in this PR, would be something to expand upon

File: HaulageApplication/HaulageAppTests/ItemTests.cs (hunk 2 above)
Comment (FMulligan) [reply HaulageApplication/HaulageAppTests/ItemTests.cs]: Will update

**Review by MxmIv — APPROVED** [review]

File: HaulageApplication/HaulageAppTests/ItemTests.cs
Comment (FMulligan) [reply HaulageApplication/HaulageAppTests/ItemTests.cs:1]: (This was meant to be a comment) Same as previous, not adding in this PR.

File: HaulageApplication/HaulageAppTests/AllItemsTests.cs
Comment (FMulligan) [reply HaulageApplication/HaulageAppTests/AllItemsTests.cs:1]: This test doesn't work with the current implementation of MockDb, and would require extensive time to look into so leaving for this PR

File: HaulageApplication/HaulageApp/ViewModels/AllItemsViewModel.cs
Comment (FMulligan) [reply HaulageApplication/HaulageApp/ViewModels/AllItemsViewModel.cs:1]: The navigation is only one line and therefore doesn't justify moving out of the class. Maybe in future could use a service for this to implement throughout the app.

**Pull request merged**